### PR TITLE
ProfileCache: prepare_profile_cache checks for valid object template

### LINKED ISSUE #126: test execution stalled if the associated pan template is not an object template
Writing a unit test, I made a mistake and declared the pan template associated with the test (declared in `use Quattor qw()`) as a normal template rather than an object template. The result was that the test execution was blocked in or after the compilation (the last message displayed was `DEBUG: Pan compiler called...` and the test was never started (if the first line was a `$this_app->info(), nothing was displayed). I check that executed the reported pan command did complete successfully.

I don't know what could be done but it would be good to exit with an error rather than getting the test stucked when such a mistake is done.

### PR DESCRIPTION
**Pull request by stdweird**
Depends on #125 
Fixes #126

### CODE REVIEW

File: build-scripts/src/main/perl/Test/Quattor/Panc.pm [hunk 1]
```diff
@@ -126,6 +126,38 @@ sub get_panc_includepath
...
+=cut
+
+sub is_object_template
+{
+    my ($profile, $resourcesdir) = @_;
+
+    $profile .= ".pan" if ($profile !~ m/\.pan$/ );
+    $profile = "$resourcesdir/$profile" if $resourcesdir && $profile !~ m/^\//;
+    my $ok;
+    open(my $TPL, '<', $profile) or croak("is_object_template failed to open $profile: $!");
```
Comment (jouvin) [inline build-scripts/src/main/perl/Test/Quattor/Panc.pm]: I'd suggest `is_object_template() failed...` to make clearer in the message that a function failed to do something...

**Comment by jouvin** [comment]
Thanks for this. I have one minor comment: I think it'd be good to have a doc somewhere (can be on the web site and thus be done outside this PR) documenting how `Perl::Critic::Policy::Quattor::UseCAFProcess` is supposed to be used.

Also I'd suggest to change the title to something reflecting more the contents of the PR where the `prepare_profile_cache()` checks seem only one part of the PR...

File: build-scripts/src/main/perl/Test/Quattor/Panc.pm (hunk 1 above)
Comment (stdweird) [inline build-scripts/src/main/perl/Test/Quattor/Panc.pm]: hmm. i originially tought that open would be the exception and thus used `croak`, but now that i think about, a type in the `use Quattor qw()` might be a lot more common, so i'll add a additional check for the existence of the file with proper clean handling. if the open then still fails, `croak` provides plenty of information

**Comment by stdweird** [comment]
@jouvin wrt more code than title, this is based on #125. i'll rebase.
wrt to using critic, once new maven tools are released, we can switch to tqu style testing, see eg https://github.com/quattor/CCM/pull/160, which includes critic.

**Comment by stdweird** [comment]
@jrha another one best included before new build-tools are released

**Pull request merged**